### PR TITLE
Add `no-nested-tests` rule (fixes #131)

### LINKED ISSUE #131: `no-nested-tests` rule
Forbid the use of `test()` inside of another `test()` call.
Linked to https://github.com/avajs/ava/issues/948

### PR DESCRIPTION
**Pull request by jfmengels**
Add `no-nested-tests` rule (fixes #131)

### CODE REVIEW

File: docs/rules/no-nested-tests.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Ensure no tests are nested
+
+Translations: [Français](https://github.com/avajs/ava-docs/blob/master/fr_FR/related/eslint-plugin-ava/docs/rules/no-nested-tests.md)
+
+In AVA, you can not nest tests, i.e. create tests inside of other tests. Doing so will lead to odd behavior.
```
Comment (sindresorhus) [inline docs/rules/no-nested-tests.md]: can not => cannot

`i.e.` => `for example,`

File: docs/rules/no-nested-tests.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+
+## Fail
+
+```js
+import test from 'ava';
+
+test('foo', t => {
+	const result = foo();
+	t.true(result.foo === 'foo');
```
Comment (sindresorhus) [inline docs/rules/no-nested-tests.md]: Drop the ` === 'foo'` part

File: docs/rules/no-nested-tests.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+	test('bar', t => {
+		t.true(result.bar === 'bar');
```
Comment (sindresorhus) [inline docs/rules/no-nested-tests.md]: Same here.

File: create-ava-rule.js [hunk 4]
```diff
@@ -82,11 +82,9 @@ module.exports = () => {
 			}
 		},
 		CallExpression: node => {
-			if (!currentTestNode) {
```
Comment (sindresorhus) [inline create-ava-rule.js]: Why was this removed? Will it have any side-effects for any of the other rules?

File: docs/rules/no-nested-tests.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
+# Ensure no tests are nested
+
+Translations: [Français](https://github.com/avajs/ava-docs/blob/master/fr_FR/related/eslint-plugin-ava/docs/rules/no-nested-tests.md)
```
Comment (sindresorhus) [inline docs/rules/no-nested-tests.md]: I assume this will be added after these rules have landed?

File: create-ava-rule.js (hunk 4 above)
Comment (jfmengels) [inline create-ava-rule.js]: It was a safe-guard (or rather optimization?) to prevent entering a new test when you were already in one.

It will only be able to make a difference for other rules (but probably won't) if you have nested tests. I can improve it to make sure the other rules won't be affected, but I'm not sure that it's worth it as there is a hard stance against nested tests. Let me know what you prefer.

File: docs/rules/no-nested-tests.md (hunk 5 above)
Comment (jfmengels) [inline docs/rules/no-nested-tests.md]: Yeah, I assumed that @forresst would work on it, as he has always been very quick on adding this. I just thought I'd make his job easier :sweat_smile:.

Comment (sindresorhus) [inline docs/rules/no-nested-tests.md]: 👌

File: create-ava-rule.js (hunk 4 above)
Comment (sindresorhus) [inline create-ava-rule.js]: It's fine like this. Was just curious about the reason.

**Comment by sindresorhus** [comment]
Thanks for all the hard work on these new rules. Can't believe we're at 23 rules already. 🤗

giphy

**Comment by jfmengels** [comment]
chris-pratt-youre-welcome-3oEduGRYSthHoPSTGU

File: docs/rules/no-nested-tests.md (hunk 5 above)
Comment (forresst) [inline docs/rules/no-nested-tests.md]: @sindresorhus @jfmengels Done : avajs/ava-docs@e779c852f2e0d81df0d485ce6481c63d7152b916 :smile_cat:

Comment (jfmengels) [inline docs/rules/no-nested-tests.md]: Man... Even quicker than I thought. Awesome! :D
